e2e-report/utils/data.js

```diff
@@ -0,0 +1,42 @@
+import fileData from '@/data/test-results.json'
+
+const nonEmptySuites = fileData.results.filter((suite) => suite.testCases?.length > 0)
+nonEmptySuites.forEach((suite) => {
+  const actualFailed = suite.testCases?.filter((t) => t.status === 'failed') || []
+  if (actualFailed.length !== suite.failed) {
+    console.warn(
+      `In suite "${suite.name}", failed value is ${suite.failed} but count of actual failed cases found is ${actualFailed.length}`,
+    )
```

Comment:
so you're saying we could just ignore the "failed" count for a suite and always set it based on the actual testCases and their status? (which I now do if there's a mismatch)

Reply:
Not quite — I'm saying we could just change the logic in the script that spits out `test-results.json` to compute the suite total fields instead of returning the raw JUnit test suite totals as is. Then you could remove this logic.

At this point we're effectively not using those totals at all, so might as well set them to useful values.

(I can do this as a follow-up PR.)